multimodal/src/autogluon/multimodal/data/utils.py

```diff
@@ -258,3 +259,83 @@ def normalize_txt(text: str) -> str:
     )
     text = unidecode(text)
     return text
+
+
+def process_ner_annotations(ner_annotations, ner_text, tokenizer, is_eval=False):
+    """
+    Generate token-level/word-level labels with given text and NER annotations.
+
+    Parameters
+    ----------
+    ner_annotations
+        The NER annotations.
+    ner_text
+        The corresponding raw text.
+    tokenizer
+        The tokenizer to be used.
+    is_eval
+        Whether it is for evaluation or not, default: False
+
+    Returns
+    -------
+    Token-level labels or word-level lavels
+    """
+    col_tokens, token_to_word_mappings, word_offsets = tokenize_ner_text(ner_text, tokenizer)
+    num_words = len(set(token_to_word_mappings)) - 1
+    word_label = [1] * num_words
+    for idx, word_offset in enumerate(word_offsets[:num_words, :]):
+        for annot in ner_annotations:
```

Comment:
Potentially optimize this function via binary search.

Reply:
May add a TODO. You can do it in a separate PR.

Reply:
Sure, I will add a TODO here.